cmd/ocm/gcp/gcp-client-shim.go

```diff
@@ -139,6 +153,38 @@ func (c *shim) CreateWorkloadIdentityProvider(
 		}
 	}
 
+	var needsUpdate bool
```

Comment:
below is needed only if the prev if check at #128 fails.

So I think you need a `return nil ` after line#149 , so it doesnt get into the below